src/vs/workbench/contrib/debug/browser/debugService.ts

```diff
@@ -452,6 +454,7 @@ export class DebugService implements IDebugService {
 			await this.launchOrAttachToSession(session);
 
 			const internalConsoleOptions = session.configuration.internalConsoleOptions || this.configurationService.getValue<IDebugConfiguration>('debug').internalConsoleOptions;
+			this.replWasOpened = this.layoutService.isVisible(Parts.PANEL_PART);
```

Comment:
replWasOpened should only be set to true if you open the repl.
False otherwise.
This way you are setting it to early, and it can happen that it is true even though you do not open the repl